modules/em/test/test_logsumexp.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import print_function
+import IMP
+import IMP.test
+import IMP.em
+import os, sys
+import numpy as np
+
+def add_logabssumprodexp(a, b):
+    s = - np.Inf
+    ss = -1
+    for i in range(len(a)):
+        s, ss = IMP.em.BayesEM3D().logabssumprodexp(s, a[i], ss, b[i])
```

Comment:
If this were a free function you would use the much less ugly `IMP.em.logabssumprodexp()` here.